core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts

```diff
@@ -16,219 +16,214 @@
+        this.languageUtilService.getAudioLanguageDescription(
+          languageCode));
+    });
+    return languageDescriptions;
+  }
+
+  ngOnInit(): void {
+    this.profilePictureDataUrl = null;
+    this.username = null;
+    this.userInfoIsLoading = true;
+    this.userIsLoggedIn = false;
+    this.userIsReviewer = false;
+    this.userCanReviewTranslationSuggestionsInLanguages = [];
+    this.userCanReviewVoiceoverSuggestionsInLanguages = [];
+    this.userCanReviewQuestions = false;
+    this.defaultHeaderVisible = true;
+
+    const prevSelectedTopicName = (
+      this.localStorageService.getLastSelectedTranslationTopicName());
+
+    this.windowRef.nativeWindow.addEventListener('scroll', () => {
+      this.scrollFunction();
+    });
+
+    this.userService.getUserContributionRightsDataAsync().then(
+      (userContributionRights) => {
+        this.userCanReviewTranslationSuggestionsInLanguages = (
+          this.getLanguageDescriptions(
+            userContributionRights
+              .can_review_translation_for_language_codes));
+
+        this.userCanReviewVoiceoverSuggestionsInLanguages = (
+          this.getLanguageDescriptions(
+            userContributionRights
+              .can_review_voiceover_for_language_codes));
+
+        this.userCanReviewQuestions = (
+          userContributionRights.can_review_questions);
+
+        this.userIsReviewer = (
+          this.userCanReviewTranslationSuggestionsInLanguages
+            .length > 0 ||
+          this.userCanReviewVoiceoverSuggestionsInLanguages
+            .length > 0 ||
+          this.userCanReviewQuestions);
+
+        this.tabsDetails.submitQuestionTab.enabled = (
+          userContributionRights.can_suggest_questions);
+      });
+
+    this.userService.getUserInfoAsync().then((userInfo) => {
+      this.userInfoIsLoading = false;
+      if (userInfo.isLoggedIn()) {
+        this.userIsLoggedIn = true;
+        this.username = userInfo.getUsername();
+      } else {
+        this.userIsLoggedIn = false;
+        this.username = '';
+      }
+    });
+
+    this.userService.getProfileImageDataUrlAsync().then(
+      (dataUrl) => {
+        this.profilePictureDataUrl = this.svgSanitizerService
+          .getTrustedSvgResourceUrl(dataUrl);
+      });
+
+    this.topicName = (
+      ContributorDashboardConstants.DEFAULT_OPPORTUNITY_TOPIC_NAME);
+    this.translationTopicService.setActiveTopicName(this.topicName);
+
+    this.contributionOpportunitiesService.getTranslatableTopicNamesAsync()
+      .then((topicNames) => {
+        if (topicNames.indexOf(prevSelectedTopicName) !== -1) {
+          this.topicName = prevSelectedTopicName;
+          this.translationTopicService.setActiveTopicName(this.topicName);
+        }
+      });
+
+    this.activeTabName = 'myContributionTab';
+
+    this.tabsDetails = {
+      ...ContributorDashboardConstants.CONTRIBUTOR_DASHBOARD_TABS_DETAILS
+    } as unknown as ContributorDashboardTabsDetails;
```

Comment:
Why is `unknown` necessary here?

Reply:
otherwise, this is throwing a read-only error, Jeevesh Garg will correct this in his second milestone as per as my conversation with him.